src/Cafe/OS/libs/nsyshid/BackendLibusb.cpp

```diff
@@ -694,22 +694,58 @@ namespace nsyshid::backend::libusb
 			return false;
 		}
 
-		// ToDo: implement this
-#if 0
-		// is this correct? Discarding "ifIndex" seems like a bad idea
-		int ret = libusb_set_configuration(handleLock->getHandle(), protocol);
-		if (ret == 0) {
-			cemuLog_logDebug(LogType::Force,
-							 "nsyshid::DeviceLibusb::setProtocol(): success");
-			return true;
+		struct libusb_config_descriptor* conf = nullptr;
+		libusb_device* dev = libusb_get_device(handleLock->GetHandle());
+		int getConfig = libusb_get_active_config_descriptor(dev, &conf);
+		if (getConfig == LIBUSB_SUCCESS)
+		{
+			for (uint8 i = 0; i < conf->bNumInterfaces; ++i)
+			{
+				int releaseSuccess = libusb_release_interface(handleLock->GetHandle(), i);
+				if (releaseSuccess < LIBUSB_SUCCESS)
+				{
+					cemuLog_logDebug(LogType::Force, "nsyshid::DeviceLibusb::SetProtocol(): Failed to release interface %i", i);
+					return false;
```

Comment:
Returning here is problematic: We still need to tell libusb to free the struct that `conf` points to!
Since you are only using `conf->bNumInterfaces` from that struct, how about calling `libusb_get_active_config_descriptor()`, making a copy of `conf->bNumInterfaces` and then immediately calling `libusb_free_config_descriptor()`?